Intersect.Server/Entities/Entity.cs

```diff
@@ -1413,8 +1530,8 @@ byte projectileDir
                 return;
             }
 
-            //If there is a knockback, knock them backwards and make sure its linear (diagonal player movement not coded).
-            if (projectile.Knockback > 0 && projectileDir < 4 && !target.Immunities.Contains(StatusTypes.Knockback))
+            // If there is knock-back: knock them backwards.
+            if (projectile.Knockback > 0 && !target.Immunities.Contains(StatusTypes.Knockback))
```

Comment:
I think you need `(projectileDir < 4 || diagonal enabled)`

Reply:
`(projectileDir < Options.Instance.MapOpts.MovementDirections)` 
might be shorter, this is either < 4 or < 8 depending on the user preferences for DM

ps. diagonal knockback _works_ fine from what've tested so far

Reply:
yeah that would also work